variables.tf

```
@@ -11,309 +11,141 @@
-  #   },
-  #   {
-  #     availability_zone = "us-east-1c",
-  #     cidr_block        = "10.0.96.0/21",
-  #   }
-  # ]
-  default = []
-}
-
-variable "private_subnets" {
-  description = "A map of private subnets to create for this VPC."
-  type        = list(map(string))
-
-  # Example:
-  #
-  # private_subnets = [
-  #   {
-  #     availability_zone = "us-east-1a",
-  #     cidr_block        = "10.0.112.0/21",
-  #   },
-  #   {
-  #     availability_zone = "us-east-1b",
-  #     cidr_block        = "10.0.128.0/21",
-  #   },
-  #   {
-  #     availability_zone = "us-east-1c",
-  #     cidr_block        = "10.0.144.0/21",
-  #   }
-  # ]
-  default = []
-}
-
-variable "intra_subnets" {
-  description = "A map of private intra subnets to create for this VPC."
-  type        = list(map(string))
-
-  # Example:
-  #
-  # intra_subnets = [
-  #   {
-  #     availability_zone = "us-east-1a",
-  #     cidr_block        = "10.0.160.0/21",
-  #   },
-  #   {
-  #     availability_zone = "us-east-1b",
-  #     cidr_block        = "10.0.176.0/21",
-  #   }
-  # ]
-  default = []
+variable "subnets" {
+  description = "(Optional) List of subnet definitions. See README.md for details. Default is []."
+  type        = any
+  default     = []
 }
 
-variable "tags" {
-  description = "A map of tags to apply to all created resources that support tags."
-  type        = map(string)
-
-  # Example:
-  #
-  # tags = {
-  #   CreatedAt = "2020-02-07",
-  #   Alice     = "Bob
-  # }
-  default = {}
+variable "nat_gateway_mode" {
+  description = "(Optional) Set the mode for the NAT Gateways. Possible inputs are \"none\" ( create not NAT Gateways at all ), \"single\" ( create a single Nat Gateway inside the first defined Public Subnet) and \"one_per_az\" ( Create one Nat Gatway inside the first Public Subnet in each availability zone ). Default is \"single\""
+  type        = string
+  default     = "single"
 }
 
 variable "eip_tags" {
-  description = "A map of tags to apply to the created NAT Gateway Elastic IP Addresses."
+  description = "(Optional) A map of tags to apply to the created NAT Gateway Elastic IP Addresses. Default is {}."
   type        = map(string)
-
-  # Example:
-  #
-  # tags = {
-  #   CreatedAt = "2020-02-07",
-  #   Alice     = "Bob
-  # }
-  default = {}
+  default     = {}
 }
 
-variable "endpoint_tags" {
-  description = "A map of tags to apply to the created VPC endpoints."
+variable "internet_gateway_tags" {
+  description = "(Optional) A map of tags to apply to the created Internet Gateway. Default is {}."
   type        = map(string)
-
-  # Example:
-  #
-  # tags = {
-  #   CreatedAt = "2020-02-07",
-  #   Alice     = "Bob
-  # }
-  default = {}
+  default     = {}
 }
 
-variable "internet_gateway_tags" {
-  description = "A map of tags to apply to the created Internet Gateway."
+variable "subnet_tags" {
+  description = "(Optional) A map of tags to apply to the vpc resource. Default is {}."
```

```suggestion
  description = "(Optional) A map of tags to apply to the created Subnet. Default is {}."
```

@mariux is the subnet reference clear here? Private & Public Subnet tags are defined explicitly below.

---

yes... one for all subnets, and special tags for private/public classes only